.github/scripts/release.py

```diff
@@ -148,96 +260,140 @@ def createGitRelease(self):
-
         with open(path_manifest, 'r') as fp:
             manifest_data = fp.read()
         yml = load(manifest_data, Loader=Loader)
         assert 'dependencies' in yml, 'Manifest YML parsing error'
+
+        # Update all the submodules per yml
+        logIndentPush()
         for dep in yml['dependencies']:
             assert 'version' in dep, 'Failed to parse submodule tag from manifest'
             assert 'repository' in dep and 'path' in dep['repository'], 'Failed to parse submodule path from manifest'
             submodule_path = dep['repository']['path']
             submodule_tag  = dep['version']
 
             # Update the submodule to point to version noted in manifest file
-            update_submodule_pointer(self.repo_path, submodule_path, submodule_tag)
+            info('%-20s : %s' % (dep['name'], submodule_tag))
+            self.updateSubmodulePointer(os.path.join(self.repo_path, submodule_path), submodule_tag)
+        logIndentPop()
 
-        commit_git_tree_changes(self.repo_path, commit_message=self.commit_msg_prefix
-                                                               + 'Bump submodules per manifest.yml for V%s' % self.version)
+        self.commitChanges(self.commit_msg_prefix + 'Bump submodules per manifest.yml for V%s' % self.version)
 
     def createReleaseZip(self):
         '''
         At the moment, the only asset we upload is the
         '''
-        remote_name = self.getRemoteEndpoint(self.repo_name)
+        zip_name = 'FreeRTOSv%s' % self.version
+        info('Packaging "%s"' % zip_name)
+        logIndentPush()
 
         # This path name is retained in zip, so we don't name it 'tmp-*' but rather keep it consistent with previous
         # packaging
-        repo_name = 'FreeRTOSv%s' % self.version
-        zip_root_path = repo_name
-        rel_repo_path = os.path.join(zip_root_path, repo_name)
+        rel_repo_path = zip_name
 
         # Clean up any old work from previous runs
-        if os.path.exists(zip_root_path):
-            shutil.rmtree(zip_root_path)
-
-        # To keep consistent with previous packages
-        os.mkdir(zip_root_path)
-
-        # Download master:HEAD. Update its file header versions and kernel submodule pointer
-        repo_path = download_git_tree(remote_name, '.', rel_repo_path, 'master', self.tag, recurse=True)
-        assert repo_path != None, 'Failed to download git tree'
-
-        self.zip = create_package(zip_root_path,
-                                  rel_repo_path,
-                                  'FreeRTOSv%s' % self.version,
-                                  exclude_files=FREERTOS_RELATIVE_FILE_EXCLUDES)
+        if os.path.exists(rel_repo_path):
+            shutil.rmtree(rel_repo_path)
+
+        # Download a fresh copy for packaging
+        info('Downloading fresh copy of %s for packing...' % zip_name, end='')
+        packaged_repo = Repo.clone_from(self.getRemoteEndpoint(self.repo_name),
```

Comment:
--depth 1 will decrease the amount of stuff cloned, could be worth investigating